### PR TITLE
Adds `--debug` flag to conversion script

### PR DESCRIPTION
**Pull request by schuylermartin45**
Adds a `--debug` (and `-d`) flag to the conversion script. This flag prints debugging output to `STDERR`. For now it prints the initial parse-tree and the post-conversion parse-tree. 

I've been using this strategy informally for months to determine if a bug is caused by an incorrect parse tree or a failure of the rendering function.

This PR also raises the unit testing coverage minimum to 80%. This should have been raised when the project was ported to conda-incubator.

Here's a sample debug output of a parser instance:
```
--------------------
RecipeParser Instance
- Variables Table:
{
  "name": "boto",
  "version": "2.49.0"
}
- Selectors Table:
  [py2k]
    - asadmin -h -> /test/commands/0
    - taskadmin -h -> /test/commands/2
- is_modified?: False
- Tree:
/
  |- package
    |- name
      |- {{ name|lower }}
    |- version
      |- {{ version }}
  |- source
    |- fn
      |- {{ name }}-{{ version }}.tar.gz
    |- url
      |- https://pypi.org/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
    |- sha256
      |- ea0d3b40a2d852767be77ca343b58a9e3a4b00d9db440efb8da74b4e58025e5a
  |- requirements
    |- host
      |- python
    |- run
      |- python
  |- build
    |- number
      |- 0
    |- script
      |- python setup.py install
  |- test
    |- commands
      |- asadmin -h
      |- s3put -h
      |- taskadmin -h
    |- imports
      |- boto
  |- about
    |- home
      |- https://github.com/boto/boto/
    |- license
      |- MIT
    |- summary
      |- Amazon Web Services Library
    |- description
      |- ['Boto aims to support the full breadth and depth of Amazon Web Services.', 'NOTE: Boto3, the next version of Boto, is stable and recommended for', 'general use.']
    |- doc_url
      |- http://docs.pythonboto.org/en/latest/
    |- doc_source_url
      |- https://github.com/boto/boto/blob/develop/docs/source/index.rst
    |- dev_url
      |- https://github.com/boto/boto/
--------------------


```

### CODE REVIEW

File: conda_recipe_manager/parser/recipe_parser_convert.py [hunk 1]
```diff
@@ -447,6 +447,10 @@ def render_to_new_recipe_format(self) -> tuple[str, MessageTable]:
           - https://github.com/conda-incubator/ceps/blob/main/cep-14.md
 
         (As of writing there is no official name other than "the new recipe format")
+        :returns: Returns a tuple containing:
+            - The converted recipe, as a string
+            - A MessageTbl instance that contains error logging
```
Comment (beeankha) [inline conda_recipe_manager/parser/recipe_parser_convert.py:452]:
```suggestion
            - A `MessageTbl` instance that contains error logging
```

Just a little nitpick here for consistency's sake

**Review by beeankha — APPROVED** [review]
